apps/web-staking/src/app/components/dropdown/DropdownText.tsx

```diff
@@ -0,0 +1,107 @@
+import {Dispatch, MutableRefObject, ReactNode, SetStateAction, useEffect, useRef} from "react";
+import { DropdownArrow } from "../../../../../../packages/ui/src/rebrand/icons/DropArrowIcon";
+import { WarningIcon } from "../../../../../../packages/ui/src/rebrand/icons/WarningIcon";
+
+
+
+const DROPDOWN_ITEMS_WITHOUT_SCROLL = 11;
+interface DropdownProps {
+    setIsOpen: (isOpen: boolean) => void;
+    isOpen: boolean;
+    selectedValue: string | null;
+    selectedValueRender: string | null;
+    setSelectedValue: Dispatch<SetStateAction<string | null>>;
+    getPreferableItems?: () => JSX.Element[];
+    getDropdownItems: () => JSX.Element[];
+    dropdownOptionsCount: number; // just put here arr.length
+    isInvalid?: boolean;
+    defaultValue?: string;
+    extraClasses?: {
+        dropdownContainer?: string;
+        dropdown?: string;
+        dropdownOptions?: string;
+    }
+}
+
+interface DropdownItemProps {
+    onClick: () => void;
+    extraClasses?: string;
+    dropdownOptionsCount: number; // just put here arr.length
```

Comment:
Why is the array length needed to be passed into the component like this. Should this not just be a bool value for the last item then ?

Reply:
Various options were tried, this one turned out to be the most optimal

Reply:
I don't understand. Please review the code and confirm that a boolean value is not sufficient. If it is, plase update the code.

Reply:
Droptown doesn't work with this boolean